controllers/chat/CChat.ts

```diff
@@ -0,0 +1,318 @@
+
+  constructor() {
+    if (CChat._exit) {
+      return CChat._exit;
+    }
+    CChat._exit = this;
+
+    if (document.location.pathname == "/chats" && this.fatalUser == null) {
+      this.getUserData();
+      this.addChatList();
+      this.sendMessage();
+    }
+    this.display(this.messageSelectDisplay);
+  }
+
+  searchUser2(userLogin: string) {
+    this.CRequests.requestSerchUsers(userLogin).then((users: [User]) => {
+      let usersHTML: string = this.searchUser.makeUser(users);
+      if (this.userList) {
+        //----view
+        this.userList.innerHTML = usersHTML;
+        this.eventAddUserToChat();
+      }
+    });
+  }
+
+  display(displayElement: HTMLDivElement | null) {
+    if (
+      this.chatDisplay &&
+      this.messageSelectDisplay &&
+      this.userManagementDisplay
+    ) {
+      this.chatDisplay.style.display = "none";
+      this.messageSelectDisplay.style.display = "none";
+      this.userManagementDisplay.style.display = "none";
+    }
+    if (displayElement) {
+      displayElement.style.display = "flex";
+    }
+  }
+
+  makeChatList(chats: any) {
+    let chatList: string = "";
+    chats.map((chat: ChatUT) => {
+      chatList += this.chatUnit.makeChatUnit(
+        chat.title,
+        chat.avatar,
+        chat.unread_count,
+        chat.last_message,
+        chat.id
+      );
+    });
+    return chatList;
+  }
+
+  addChatList() {
+    this.display(this.messageSelectDisplay);
+    this.CRequests.requestGetChats()
+      .then((chats) => {
+        const parent = document.querySelector(this.parentSelector);
+        if (parent) {
+          parent.innerHTML = this.makeChatList(chats);
+          this.addEventChat();
+        }
+      })
+      .catch((err: string) => {
+        console.error("error", err);
+      });
+  }
+
+  addNewСhat(chatName: string) {
+    this.CRequests.requestAddNewChat(chatName)
+      .then(() => {
+        this.addChatList();
+      })
+      .catch((err: string) => {
+        console.error("error", err);
+      });
+  }
+
+  addEventChat() {
+    let CL = document.querySelectorAll(this.elementSelector);
+    CL.forEach((chat: HTMLDivElement) => {
+      if (chat) {
+        this.addEventDelChat(chat);
+        this.addEventSerchUsersPanel(chat);
+      }
+      chat.addEventListener("click", () => {
+        if (this.chatDisplay && this.messageSelectDisplay) {
+          if (this.activeChat != chat) {
+            if (this.activeChat != null) {
+              this.activeChat.style.border = "";
+            }
+            this.activeChat = chat;
+            chat.style.border = "1px solid silver";
+            this.messageSelectDisplay.style.display = "none";
+            this.chatID = chat.getAttribute("data-id");
+            this.startSocket();
+            this.display(this.chatDisplay);
+          } else {
+            chat.style.border = "";
+            this.display(this.messageSelectDisplay);
+            if (this.activeChat != null) {
```

Comment:
Такие условия обычно принято сокращать до 
```suggestion
            if (!this.activeChat) {
```